src/qt/walletview.cpp

```diff
@@ -222,6 +225,69 @@ void WalletView::gotoVerifyMessageTab(QString addr)
         signVerifyMessageDialog->setAddress_VM(addr);
 }
 
+void WalletView::gotoLoadPSBT()
+{
+    QString filename = GUIUtil::getOpenFileName(this,
+        tr("Load Transaction Data"), QString(),
+        tr("Partially Signed Transaction (*.psbt)"), nullptr);
+    if (filename.isEmpty()) return;
+    std::ifstream in(filename.toLocal8Bit().data(), std::ios::binary);
+    // https://stackoverflow.com/questions/116038/what-is-the-best-way-to-read-an-entire-file-into-a-stdstring-in-c
+    std::string data(std::istreambuf_iterator<char>{in}, {});
+
+    std::string error;
+    PartiallySignedTransaction psbtx;
+    if (!DecodeRawPSBT(psbtx, data, error)) {
+        Q_EMIT message(tr("Error"), tr("Unable to decode PSBT file") + "\n" + QString::fromStdString(error), CClientUIInterface::MSG_ERROR);
+        return;
+    }
+
+    PSBTAnalysis analysis = AnalyzePSBT(psbtx);
+    bool have_all_sigs = (analysis.next == PSBTRole::FINALIZER) || (analysis.next == PSBTRole::EXTRACTOR);
+
+    CMutableTransaction mtx;
+    bool complete = FinalizeAndExtractPSBT(psbtx, mtx);
+
+    QMessageBox msgBox;
+    msgBox.setText("Partially Signed Bitcoin Transaction");
```

Comment:
"Completed"? Might want to be very obvious to user that there's nothing else to be done.

Reply:
That's a good point re: wording, but I would suggest not bikeshedding on it because it's going to get changed again in my #18027 .

Reply:
It's not necessarily complete. The `setInformativeText` below covers the various options. But I changed it to "PSBT" to hopefully reduce confusion.